### PR TITLE
GO-1625 Support spaceID in full-text search

### PR DESCRIPTION
**Pull request by KirillSto**

### CODE REVIEW

File: pkg/lib/localstore/ftsearch/ftsearch.go [hunk 1]
```diff
@@ -174,8 +175,17 @@ func (f *ftSearch) getTerms(qry string) []string {
...
+	if spaceID != "" {
+		spaceQuery := bleve.NewMatchQuery(spaceID)
+		rootQuery = bleve.NewConjunctionQuery(disjunctionQuery, spaceQuery)
+	} else {
+		rootQuery = disjunctionQuery
+	}
+
+	searchRequest := bleve.NewSearchRequest(rootQuery)
 	searchRequest.Size = 100
 	searchRequest.Explain = true
```
Comment (fat-fellow) [inline pkg/lib/localstore/ftsearch/ftsearch.go:190]: tests for fts please

Comment (KirillSto) [reply pkg/lib/localstore/ftsearch/ftsearch.go:190]: Will be discussed and implemented under https://linear.app/anytype/issue/GO-1734/graylog-should-have-ability-to-write-the-file

File: pkg/lib/localstore/ftsearch/ftsearch.go [hunk 2]
```diff
@@ -174,8 +175,17 @@ func (f *ftSearch) getTerms(qry string) []string {
 	return terms
 }
 
-func (f *ftSearch) doSearch(queries []query.Query) (results []string, err error) {
-	searchRequest := bleve.NewSearchRequest(bleve.NewDisjunctionQuery(queries...))
+func (f *ftSearch) doSearch(spaceID string, queries []query.Query) (results []string, err error) {
+	var rootQuery query.Query
+	disjunctionQuery := bleve.NewDisjunctionQuery(queries...)
```
Comment (AnastasiaShemyakinskaya) [inline pkg/lib/localstore/ftsearch/ftsearch.go:180]: AFAIU 179-188 can be simplified
```
rootQuery := bleve.NewDisjunctionQuery(queries...)
if spaceID != "" {
		spaceQuery := bleve.NewMatchQuery(spaceID)
		rootQuery = bleve.NewConjunctionQuery(disjunctionQuery, spaceQuery)
}
```

File: pkg/lib/localstore/objectstore/queries.go [hunk 3]
```diff
@@ -108,6 +108,22 @@ func (s *dsObjectStore) makeFTSQuery(text string, filters *database.Filters) (*d
...
 }
 
+func getSpaceIDFromFilter(fltr filter.Filter) (spaceID string) {
+	switch f := fltr.(type) {
+	case filter.Eq:
+		if f.Key == bundle.RelationKeySpaceId.String() {
+			return f.Value.GetStringValue()
+		}
+	case filter.AndFilters:
+		for _, af := range f {
```
Comment (AnastasiaShemyakinskaya) [inline pkg/lib/localstore/objectstore/queries.go:118]: May be 118-122 extract to separate function to reduce nested blocks

File: pkg/lib/localstore/ftsearch/ftsearch.go (hunk 2 above)
Comment (KirillSto) [reply pkg/lib/localstore/ftsearch/ftsearch.go]: Thank you! Simplified the code by getting rid of disjunctionQuery variable

File: pkg/lib/localstore/objectstore/queries.go (hunk 3 above)
Comment (KirillSto) [reply pkg/lib/localstore/objectstore/queries.go]: Reduced nested blocks by introducing separate func for AndFilters iteration

File: pkg/lib/localstore/ftsearch/ftsearch.go (hunk 1 above)
Comment (KirillSto) [reply pkg/lib/localstore/ftsearch/ftsearch.go:187]: Added unit tests for multi space

**Review by AnastasiaShemyakinskaya — APPROVED** [review]

**Review by fat-fellow — APPROVED** [review]

**Pull request merged**